virttest/test_setup.py

```diff
@@ -1767,3 +1769,69 @@ def _all_killed():
             wait.wait_for(_all_killed, timeout=60)
         except OSError:
             logging.warn("egd.pl is running")
+
+
+class StraceQemu(object):
+
+    """
+    Attach strace to qemu VM processes, if enable_strace is 'yes'.
+    It's useful to analyze qemu hang issue. But it will generate
+    a large size logfile if it is enabled, so compressed original
+    logfiles after strace process terminated to safe disk space.
+    """
+
+    def __init__(self, test, params, env):
+        self.env = env
+        self.test = test
+        self.params = params
+        self.process = path.find_command("strace")
+        self.log_tarball = "strace.tgz"
+
+    @property
+    def root_dir(self):
+        root_dir = os.path.join(self.test.debugdir, "strace")
+        if not os.path.isdir(root_dir):
+            os.makedirs(root_dir)
+        return root_dir
+
+    def _generate_cmd(self, vm):
+        """Generate strace start command line"""
+        pid = vm.get_pid()
+        template = "{strace} -p {pid} -f -F -T -C -o {logfile}"
+        logfile = os.path.join(self.root_dir, "%s.log" % pid)
+        kwargs = {"strace": self.process,
+                  "pid": pid,
+                  "logfile": logfile}
+        return template.format(**kwargs)
+
+    def _start(self, cmd):
+        """Start strace process in sub-process"""
+        sub_process = process.SubProcess(cmd, shell=True)
+        sub_process.start()
+        if self.env["strace_processes"]:
```

Comment:
11:13:02 stacktrace       L0042 ERROR|   File "/root/WorkFolder/avocado/avocado/virttest/test_setup.py", line 1811, in _start
11:13:02 stacktrace       L0042 ERROR|     if self.env["strace_processes"]:
11:13:02 stacktrace       L0042 ERROR|   File "/usr/lib64/python2.7/UserDict.py", line 23, in __getitem__
11:13:02 stacktrace       L0042 ERROR|     raise KeyError(key)
11:13:02 stacktrace       L0042 ERROR| KeyError: 'strace_processes'